File: test/zenroom_exec.py
import sys
from pathlib import Path
from time import process_time

path = Path("../bindings/python3/zenroom/")
sys.path.append(path.resolve())

from zenroom.zenroom import zenroom_exec

print("[PY] zenroom_exec %s" % sys.argv[1])
script_path = Path(sys.argv[1])

try:
    script = script_path.read_text()
except UnicodeDecodeError:
    script = script_path.read_text(encoding='iso-8859-1')

start = process_time()
result = zenroom_exec(script=script)
end = process_time() - start

print(result if result else '')
print("--- %s seconds ---" % end)
print("@", "="*40, "@")

File: test/zenroom_exec_go/zenroom_exec.go
package main

import (
  "os"
  "fmt"
  "io/ioutil"
  "path"
  "strings"
  "github.com/dyne/Zenroom/bindings/golang/zenroom"
  "log"
  "time"
)

func main() {
  final_path := path.Join(os.Args[1:]...)

  fmt.Println("[GO] zenroom_exec ", final_path)

  buffer, err := ioutil.ReadFile(final_path)
  if err != nil {
    log.Fatal(err)
  }
  script := string(buffer)

  start := time.Now()
  result, success := zenroom.ZenroomExec(script, "", "", "")
  elapsed := time.Since(start)
  if !success {
    log.Fatal(result)
  }

  fmt.Println(result)
  fmt.Println("--- ", elapsed, "seconds ---")
  fmt.Println("@", strings.Repeat("=", 40), "@")

}

